tests/processors/base/writers/test_source.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import mock
+import processors.base.writers as writers
+
+
+class TestSourceWriter(object):
+    @mock.patch('processors.base.writers.source.helpers')
+    def test_skips_source_with_invalid_url(self, helpers_mock):
+        conn = _get_mock_conn()
+        conn['database']['sources'].find_one.return_value = None
+        source = mock.MagicMock(spec_set=dict)
+        source.__getitem__.return_value = 'mockingbird'
+        source.get.return_value = 'mockingbird'
```

Comment:
Isn't the `source` just a regular `dict`? In that case, there's no need to use a mock here, you can simply pass a regular dict with the values you want.